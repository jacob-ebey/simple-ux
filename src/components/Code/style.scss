.code-block {
  code {
    display: block;
    overflow-x: auto;
    padding: 10px;
    background-color: #ccc;
  }
}