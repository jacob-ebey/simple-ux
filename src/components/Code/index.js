import { h } from 'preact' // eslint-disable-line no-unused-vars

import { merge } from '../tools'

import './style.scss'

const Code = ({ class: className, children, ...rest }) => {
  return (
    <pre
      {...rest}
      class={merge('code-block', className)}
    ><code>
      {
        children && children.length === 1 && typeof children[0] === 'string'
          ? children[0].trim()
          : children
      }
    </code></pre>
  )
}

export default Code
