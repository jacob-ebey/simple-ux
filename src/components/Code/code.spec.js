/* eslint-env jest */
import { h } from 'preact' // eslint-disable-line no-unused-vars
import { shallow } from 'preact-render-spy'

import Code from './'

describe('Code', () => {
  describe('- snapshots -', () => {
    it('can render', () => {
      const wrapper = shallow(<Code />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can add custom class', () => {
      const wrapper = shallow(<Code class='custom' />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can render children child', () => {
      const wrapper = shallow(
        <Code>
          <span>Child 1</span>
          <span>Child 2</span>
        </Code>
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('can render string child', () => {
      const wrapper = shallow(
        <Code>
          {`console.log('Hello, Test!')`}
        </Code>
      )
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('- logic -', () => {
    it('trims string child', () => {
      const code = ` console.log('Hello, Test!') `
      const wrapper = shallow(
        <Code>
          {code}
        </Code>
      )

      const renderedCode = wrapper.text()

      expect(renderedCode).toBe(code.trim())
    })
  })
})
