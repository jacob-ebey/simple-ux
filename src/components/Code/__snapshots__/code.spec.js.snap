// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code - snapshots - can add custom class 1`] = `
preact-render-spy (1 nodes)
-------
<pre class="code-block custom">
  <code></code>
</pre>

`;

exports[`Code - snapshots - can render 1`] = `
preact-render-spy (1 nodes)
-------
<pre class="code-block">
  <code></code>
</pre>

`;

exports[`Code - snapshots - can render children child 1`] = `
preact-render-spy (1 nodes)
-------
<pre class="code-block">
  <code>
    <span>Child 1</span>
    <span>Child 2</span>
  </code>
</pre>

`;

exports[`Code - snapshots - can render string child 1`] = `
preact-render-spy (1 nodes)
-------
<pre class="code-block">
  <code>console.log('Hello, Test!')</code>
</pre>

`;
